docs-go/flows.md

```diff
@@ -0,0 +1,187 @@
+# Flows
+
+Flows are functions with some additional characteristics: they are strongly
+typed, streamable, locally and remotely callable, and fully observable.
+Firebase Genkit provides CLI and developer UI tooling for working with flows
+(running, debugging).
+
+## Defining flows
+
+A flow wraps a function:
+
+- {Go}
+
+  ```go
+  menuSuggestionFlow := genkit.DefineFlow(
+    "menuSuggestionFlow",
+    func(ctx context.Context, restaurantTheme string, _ genkit.NoStream) (string, error) {
+      suggestion := makeMenuItemSuggestion(restaurantTheme)
+      return suggestion, nil
+    },
+  )
+  ```
+
+You can define input and output schemas for a flow, which is useful when you
+want type-safe structured output.
+
+- {Go}
+
+  Genkit converts Go types into JSON Schema using
+  [`invopop/jsonschema`](https://pkg.go.dev/github.com/invopop/jsonschema):
+
+  ```go
+  type MenuSuggestion struct {
+    ItemName    string `json:"item_name"`
+    Description string `json:"description"`
+    Calories    int    `json:"calories"`
+  }
+  ```
+
+  ```go
+  menuSuggestionFlow := genkit.DefineFlow(
+    "menuSuggestionFlow",
+    func(ctx context.Context, restaurantTheme string, _ genkit.NoStream) (MenuSuggestion, error) {
+      suggestion := makeStructuredMenuItemSuggestion(restaurantTheme)
+      return suggestion, nil
+    },
+  )
+  ```
+
+## Running flows
+
+- {Go}
+
+  ```go
+  suggestion, err := genkit.RunFlow(context.Background(), menuSuggestionFlow, "French")
+  ```
+
+You can use the CLI to run flows as well:
+
+```posix-terminal
+genkit flow:run menuSuggestionFlow '"French"'
+```
+
+### Streamed
+
+Here's a simple example of a flow that can stream values:
+
+- {Go}
+
+  ```go
+  // Types for illustrative purposes.
+  type inputType string
+  type outputType string
+  type streamType string
+
+  menuSuggestionFlow := genkit.DefineFlow(
+    "menuSuggestionFlow",
+    func(
+      ctx context.Context,
+      restaurantTheme inputType,
+      callback func(context.Context, streamType) error,
+    ) (outputType, error) {
+      var menu outputType = ""
+      menuChunks := make(chan streamType)
+      go makeFullMenuSuggestion(restaurantTheme, menuChunks)
+      for {
+        chunk, more := <-menuChunks
```

Comment:
In Go we typically call the second variable here `ok`, meaning that the value (`chunk` in this case) is valid.  The name `more` hints that there are more values to come, which isn't quite right.